File: examples/ts/tags.ts
/**
 * Tag generation example - extract content word tags for indexing/search
 *
 * Run:
 *   npx tsx examples/ts/tags.ts
 *
 * Requires: npm package built (yarn build)
 */
import { Suzume } from '../../dist/index.js';

const suzume = await Suzume.create();

const text = '東京都渋谷区で開催されたイベントに参加しました';

// Generate all content word tags
const tags = suzume.generateTags(text);
console.log('All tags:');
for (const t of tags) {
  console.log(`  ${t.tag} (${t.pos})`);
}

// Nouns only, exclude basic words
const nounTags = suzume.generateTags(text, {
  pos: ['noun'],
  excludeBasic: true,
});
console.log('\nNoun tags (excluding basic):');
for (const t of nounTags) {
  console.log(`  ${t.tag}`);
}

// Verbs and adjectives with lemma
const verbAdjTags = suzume.generateTags(text, {
  pos: ['verb', 'adjective'],
  useLemma: true,
});
console.log('\nVerb/Adjective tags (lemma):');
for (const t of verbAdjTags) {
  console.log(`  ${t.tag} (${t.pos})`);
}

File: examples/cpp/tags.cpp
/**
 * Tag generation example - extract content word tags for indexing/search
 *
 * Build (from project root):
 *   cmake -B build && cmake --build build --parallel
 *   g++ -std=c++17 -Isrc -Lbuild/lib -o tags examples/cpp/tags.cpp \
 *     -lsuzume -lsuzume_analysis -lsuzume_postprocess -lsuzume_grammar \
 *     -lsuzume_dictionary -lsuzume_core -lsuzume_normalize -lsuzume_pretokenizer
 */
#include <iostream>

#include "core/types.h"
#include "suzume.h"

int main() {
  suzume::Suzume analyzer;

  // Generate tags (content words suitable for search indexing)
  auto tags = analyzer.generateTags("東京都渋谷区で開催されたイベントに参加しました");

  std::cout << "Tags:\n";
  for (const auto& tag : tags) {
    std::cout << "  " << tag.tag << " ("
              << suzume::core::posToString(tag.pos) << ")\n";
  }

  // With custom options: nouns only, exclude basic words
  suzume::postprocess::TagGeneratorOptions opts;
  opts.pos_filter = suzume::postprocess::kTagPosNoun;
  opts.exclude_basic = true;

  auto noun_tags = analyzer.generateTags(
      "東京都渋谷区で開催されたイベントに参加しました", opts);

  std::cout << "\nNoun tags (excluding basic):\n";
  for (const auto& tag : noun_tags) {
    std::cout << "  " << tag.tag << "\n";
  }

  return 0;
}
